internal/provider/buildEvent.go

```diff
@@ -0,0 +1,240 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/kairos-io/kairos-sdk/bus"
+	"github.com/kairos-io/kairos-sdk/types"
+	"github.com/kairos-io/kairos-sdk/utils"
+	"github.com/kairos-io/provider-kairos/v2/internal/services"
+	"github.com/mudler/go-pluggable"
+)
+
+const (
+	K3s = "k3s"
+	K0s = "k0s"
```

Comment:
This file is shouting "interfaces" all over it :D. We are probably not going to add any other implementations if we are going towards "one provider, one workload" mentality. But still, it would make this code more readable if the 2 "workloads" were separated in code.

Reply:
good point, we actually already do something like that in here https://github.com/kairos-io/provider-kairos/blob/main/internal/role/p2p/k8s.go